src/test/java/core/basesyntax/service/DataWriterServiceTest.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax.service;
+
+import core.basesyntax.service.impl.FileWriterServiceImpl;
+import core.basesyntax.service.impl.ScreenWriterServiceImpl;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class DataWriterServiceTest {
+    private static DataWriterService fileWriter;
+    private static DataWriterService incorrectFileWriter;
+    private static DataWriterService screenWriter;
+    private static List<String> dataList;
+    private static final Path TEMP_FILE_PATH = Path.of("src/main/resources/reports/temp.csv");
+    private static final String WRONG_PATH = "src/mian/resources/reports/temp.csv";
+
+    @BeforeClass
+    public static void setUp() {
+        fileWriter = new FileWriterServiceImpl(TEMP_FILE_PATH.toString());
+        incorrectFileWriter = new FileWriterServiceImpl(WRONG_PATH);
+        screenWriter = new ScreenWriterServiceImpl();
+        dataList = InitialisationService.getDataList();
+    }
+
+    @Test
+    public void writeToFile_ok() throws IOException {
+        fileWriter.writeData(dataList);
+        Assert.assertTrue(Files.exists(TEMP_FILE_PATH));
+        Files.deleteIfExists(TEMP_FILE_PATH);
```

Comment:
I think you are testing here not the data that should have been written to a file, but whether the file was created at all.